### DISCUSSION TITLE
Multiple users

### DISCUSSION THREAD

**Discussion by Venique** [shadowsocks/shadowsocks-rust#990] [Ideas]
Hello!

When multiple users via config.json will be implemented in rust port of shadowsocks?

**Comment by zonyitoo** [comment]
It was supported in AEAD-2022 ciphers.

**Reply by Venique** [reply]
Now I'm using _chacha20-ietf-poly1305_ as recommended. If I switch (for example) to _2022-blake3-chacha20-poly1305_ I'll be able to use
`"users": [{"name": "username","password": "4w0GKJ9U3Ox7CIXGU4A3LDQAqP6qrp/tUi/ilpOR9p4="}],`
in config.json, right?

If yes, which one from AEAD-2022 is better for KVM?

**Reply by database64128** [reply]
`2022-blake3-aes-128-gcm`
